services/notifications/pkg/service/persistence.go

```diff
@@ -0,0 +1,116 @@
+package service
+
+import (
+	"context"
+	"encoding/json"
+	user "github.com/cs3org/go-cs3apis/cs3/identity/user/v1beta1"
+	"github.com/owncloud/ocis/v2/ocis-pkg/log"
+	v0 "github.com/owncloud/ocis/v2/protogen/gen/ocis/messages/eventhistory/v0"
+	ehsvc "github.com/owncloud/ocis/v2/protogen/gen/ocis/services/eventhistory/v0"
+	"github.com/pkg/errors"
+	"go-micro.dev/v4/store"
+)
+
+type userEventStore struct {
+	log           log.Logger
+	store         store.Store
+	historyClient ehsvc.EventHistoryService
+}
+
+type userEventIds struct {
+	User     *user.User `json:"user"`
+	EventIds []string   `json:"event_ids"`
+}
+
+type userEvents struct {
+	User   *user.User
+	Events []*v0.Event
+}
+
+const (
+	intervalDaily  = "daily"
+	intervalWeekly = "weekly"
```

Comment:
please prefix unexported global variables with `_`. This way one can easily spot which vars a global and which are local
```suggestion
	_intervalDaily  = "daily"
	_intervalWeekly = "weekly"
```

Reply:
👍